src/pkg/cli/client/byoc/aws/list.go

```diff
@@ -0,0 +1,151 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"iter"
+	"runtime"
+	"sync"
+
+	"github.com/DefangLabs/defang/src/pkg/cli/client/byoc"
+	"github.com/DefangLabs/defang/src/pkg/clouds/aws"
+	"github.com/DefangLabs/defang/src/pkg/clouds/aws/ecs/cfn"
+	"github.com/DefangLabs/defang/src/pkg/clouds/aws/region"
+	"github.com/DefangLabs/defang/src/pkg/term"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	s3types "github.com/aws/aws-sdk-go-v2/service/s3/types"
+)
+
+func listPulumiStacksInRegion(ctx context.Context, region aws.Region) (iter.Seq[string], error) {
+	// Determine the bucket name from the CloudFormation stack outputs (if any); this is slow!
```

Comment:
Since S3 list bucket API is global, would  scan though all buckets with prefix `defang-cd` faster than trying to populate the cfn stack in each region?

Reply:
Hmm, yeah you're probably right. It's the `DescribeStacks` that is slooooooow.